addons/account/models/account_tax.py

```diff
@@ -1120,69 +1155,1033 @@ def _adapt_price_unit_to_another_taxes(self, price_unit, product, original_taxes
+                amounts['raw_tax_amount_currency'] += tax_data['raw_tax_amount_currency']
+                amounts['tax_amount'] += tax_data['tax_amount']
+                amounts['raw_tax_amount'] += tax_data['raw_tax_amount']
+                amounts['base_amount_currency'] += tax_data['base_amount_currency']
+                amounts['raw_base_amount_currency'] += tax_data['raw_base_amount_currency']
+                amounts['base_amount'] += tax_data['base_amount']
+                amounts['raw_base_amount'] += tax_data['raw_base_amount']
+                if not base_line['special_type']:
+                    amounts['base_lines'].append(base_line)
+
+        # Round 'total_per_tax'.
+        for (tax, currency, _is_refund), amounts in total_per_tax.items():
+            amounts['raw_tax_amount_currency'] = currency.round(amounts['raw_tax_amount_currency'])
+            amounts['raw_tax_amount'] = company.currency_id.round(amounts['raw_tax_amount'])
+            amounts['raw_base_amount_currency'] = currency.round(amounts['raw_base_amount_currency'])
+            amounts['raw_base_amount'] = company.currency_id.round(amounts['raw_base_amount'])
+
+        # If tax lines are provided, the totals will be aggregated according them.
+        if tax_lines:
+            manually_edited_tax_key = set()
+            for tax_line in tax_lines:
+                tax_rep = tax_line['tax_repartition_line_id']
+                sign = tax_line['sign']
+                tax = tax_rep.tax_id
+                currency = tax_line['currency_id']
+                key = (tax, currency, tax_rep.document_type == 'refund')
+
+                if key not in manually_edited_tax_key:
+                    total_per_tax[key]['raw_tax_amount_currency'] = 0.0
+                    total_per_tax[key]['raw_tax_amount'] = 0.0
+                    manually_edited_tax_key.add(key)
+                total_per_tax[key]['raw_tax_amount_currency'] += sign * tax_line['amount_currency']
+                total_per_tax[key]['raw_tax_amount'] += sign * tax_line['balance']
+
+        # Dispatch the delta accross the base lines.
+        for (tax, _currency, _is_refund), amounts in total_per_tax.items():
+            if not amounts['base_lines']:
+                continue
+
+            base_line = min(
+                amounts['base_lines'],
+                key=lambda base_line: base_line['tax_details']['total_included_currency'],
+            )
+            tax_details = base_line['tax_details']
+            index, tax_data = next((i, x) for i, x in enumerate(tax_details['taxes_data']) if x['tax'] == tax)
+            delta_base_amount_currency = amounts['raw_base_amount_currency'] - amounts['base_amount_currency']
+            delta_base_amount = amounts['raw_base_amount'] - amounts['base_amount']
+            if index == 0:
+                tax_details['delta_base_amount_currency'] += delta_base_amount_currency
+                tax_details['delta_base_amount'] += delta_base_amount
```

Comment:
As discussed I think this method of allocating the deltas could be improved (if `delta_base_amount_currency` and `delta_base_amount` have opposite signs this might result in the tax amount in invoice currency and the tax amount in company currency going opposite ways, which might fail validation in some EDIs) 
(But we also noted that all the EDIs that enforce such strong validations allow us to report tax values with a larger number of decimals, so we can use the raw amounts and round them to 4 or 6 decimals, so in practice this shouldn't be a problem)